core/src/main/java/brooklyn/util/config/ConfigBag.java

```diff
@@ -132,12 +166,12 @@ public String getDescription() {
     
     /** current values for all entries 
      * @return non-modifiable map of strings to object */
-    public Map<String,Object> getAllConfig() {
-        return Collections.unmodifiableMap(config);
+    public synchronized Map<String,Object> getAllConfig() {
```

Comment:
Preference to not synchronize on `this` - if other code synchronizes on the object then leads to a risk of deadlock. It leaks the implementation detail. Better to declare a `private final` field, and synchronise on that.